libraries/resource_provider.rb

```diff
@@ -19,35 +19,59 @@ class ResourceProvider
       :item,          # The OneviewSDK resource instance
       :sdk_api_version,  # Keep track of the API version used for the SDK resource class
       :sdk_variant,      # Keep track of the variant used for the SDK resource class
-      :sdk_resource_type # Keep track of the SDK resource class name
+      :sdk_resource_type, # Keep track of the SDK resource class name
+      :sdk_secondary_api  # Keep track of the internal secondary API used by the SDK
 
     def initialize(context)
       @context = context
       @resource_name = context.resource_name
       @name = context.name
-      name_arr = self.class.to_s.split('::')
+      klass = namespace_handler
+      c = case @sdk_secondary_api
+          when /ImageStreamer/
+            OneviewCookbook::Helper.build_image_streamer_client(context.client)
+          else
+            OneviewCookbook::Helper.build_client(context.client)
+          end
+      new_data = JSON.parse(context.data.to_json) rescue context.data
+      @item = context.property_is_set?(:api_header_version) ? klass.new(c, new_data, context.api_header_version) : klass.new(c, new_data)
+      @item['name'] ||= context.name
+    end
+
+    # Helper method that analyzes the namespace and defines the class variables and the resource class itself
+    # @return [OneviewSDK::Resource] SDK resource class
+    # @raise [NameError] If for some reason the method could not resolve the namespace to a OneviewSDK::Resource
+    def namespace_handler
```

Comment:
And just as a suggestion, maybe we can use a more descriptive name for this method. Something like `load_namespace`, `load_class_from_namespace` or `parse_namespace`? Handler makes it sound like an exception handler or something, but this method both sets data and returns important info (class reference).